### PR TITLE
Feature/customagentpool

### LINKED ISSUE #53: Function: Get/New-ADOPSElasticPool
We need to create new functions for life-cycle management of Elastic Pools in Azure DevOps.

This issue will cover the get/create/new functions.

### LINKED ISSUE #55: Get-ADOPSPool
New Function to get/list Azure DevOps Agent pools.

### LINKED ISSUE #57: Function: Get-ADOPSNode
We need a new function to get/list Azure DevOps node information.
Docs: https://docs.microsoft.com/en-us/rest/api/azure/devops/distributedtask/nodes/list?view=azure-devops-rest-7.1

### LINKED ISSUE #97: New Function to help assist create the Elastic Pool object
Create a new PowerShell function that can assist to create a baseline Elastic Pool object.
The output should be able to alternate between Json and PowerShell Custom Object.

### PR DESCRIPTION
**Pull request by SebastianClaesson**
# Contributing a Pull Request

If you haven't already, read the full [contribution guide](https://github.com/ADOPS/ADOPS/blob/main/CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, run through the relevant checklist below.

## Overview/Summary

Adding new functions to enable creation/listing of custom agent pools.
This allows you to pin an agent pool to a Azure Virtual Machine Scale Set.

## This PR fixes/adds/changes/removes

1. Closes #53 
2. Closes #57 
3. Closes #55 
4. Closes #97 

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/ADOPS/ADOPS/pulls)
- [X] Associated it with relevant [issues](https://github.com/ADOPS/ADOPS/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/ADOPS/ADOPS/tree/main)
- [X] Performed testing and provided evidence.
- [X] Updated relevant and associated documentation.

### CODE REVIEW

File: Source/Public/New-ADOPSElasticpool.ps1 [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    if (-not [string]::IsNullOrEmpty($Organization)) {
+        $Org = GetADOPSHeader -Organization $Organization
+    } else {
+        $Org = GetADOPSHeader
+        $Organization = $Org['Organization']
+    }
+
+    if ($PSBoundParameters.ContainsKey('ProjectId')) {
+        $Uri = "https://dev.azure.com/$Organization/_apis/distributedtask/elasticpools?poolName=$PoolName`&authorizeAllPipelines=$AuthorizeAllPipelines`&autoProvisionProjectPools=$AutoProvisionProjectPools&projectId={projectId}&api-version=7.1-preview.1"
```
Comment (bjompen) [inline Source/Public/New-ADOPSElasticpool.ps1:36]: I believe the `{projectID}` should be replaced with `$ProjectID` here?

File: Source/Public/New-ADOPSElasticpool.ps1 [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    } else {
+        $Uri = "https://dev.azure.com/$Organization/_apis/distributedtask/elasticpools?poolName=$PoolName`&authorizeAllPipelines=$AuthorizeAllPipelines`&autoProvisionProjectPools=$AutoProvisionProjectPools&api-version=7.1-preview.1"
+    }
+    
+    $Method = 'POST'
+    $ElasticPoolInfo = InvokeADOPSRestMethod -Uri $Uri -Method $Method -Organization $Organization -Body $Body
```
Comment (bjompen) [inline Source/Public/New-ADOPSElasticpool.ps1:42]: Can we include a default body? There are some parameters I would like to have settable, but most can be just "standard" for now. The option would be to also add a command like `New-ADOPSElasticPoolObject` or something that we could use to create a body or object we could then use as parameter value in this command.

File: Source/Public/New-ADOPSElasticpool.ps1 (hunk 1 above)
Comment (SebastianClaesson) [reply Source/Public/New-ADOPSElasticpool.ps1]: Thanks, good find :) Sometimes it goes to quick!

File: Source/Public/New-ADOPSElasticpool.ps1 (hunk 2 above)
Comment (SebastianClaesson) [reply Source/Public/New-ADOPSElasticpool.ps1:40]: I've added a new function called New-ADOPSElasticPoolObject
Let's discuss and see how we can best implement it!

**Review by bjompen — APPROVED** [review]
I like it! Great job!

**Pull request merged**